crates/e2e/tests/e2e/cow_amm.rs

```diff
@@ -605,3 +608,389 @@ async fn cow_amm_driver_support(web3: Web3) {
+        .contracts()
+        .weth
+        .balance_of(bob.address())
+        .call()
+        .await
+        .unwrap();
+
+    // Compensate a delay between the `CurrentBlockStream` and the actual onchain
+    // data.
+    tokio::time::sleep(tokio::time::Duration::from_secs(1)).await;
+
+    // Set the fees appropriately
+    let fee_cow_amm = U256::exp10(16); // 0.01 WETH
+    let fee_user = to_wei(1); // 1 DAI
+
+    let mocked_solutsion = |order_uid: OrderUid| {
+        Solution {
+            id: 1,
+            prices: HashMap::from([
+                (dai.address(), to_wei(1)),                         // 1 DAI = $1
+                (onchain.contracts().weth.address(), to_wei(2300)), // 1 WETH = $2300
+            ]),
+            trades: vec![
+                solvers_dto::solution::Trade::Jit(solvers_dto::solution::JitTrade {
+                    order: solvers_dto::solution::JitOrder {
+                        sell_token: cow_amm_order.sell_token,
+                        buy_token: cow_amm_order.buy_token,
+                        receiver: cow_amm_order.receiver.unwrap_or_default(),
+                        sell_amount: cow_amm_order.sell_amount,
+                        buy_amount: cow_amm_order.buy_amount,
+                        valid_to: cow_amm_order.valid_to,
+                        app_data: cow_amm_order.app_data.0,
+                        kind: Kind::Sell,
+                        sell_token_balance: SellTokenBalance::Erc20,
+                        buy_token_balance: BuyTokenBalance::Erc20,
+                        signing_scheme: SigningScheme::Eip1271,
+                        signature: signature.clone(),
+                    },
+                    executed_amount: cow_amm_order.sell_amount - fee_cow_amm,
+                    fee: Some(fee_cow_amm),
+                }),
+                solvers_dto::solution::Trade::Fulfillment(solvers_dto::solution::Fulfillment {
+                    order: order_uid.0,
+                    executed_amount: executed_amount - fee_user,
+                    fee: Some(fee_user),
+                }),
+            ],
+            pre_interactions: vec![cow_amm_commitment.clone()],
+            interactions: vec![],
+            post_interactions: vec![],
+            gas: None,
+        }
+    };
+
+    // Configure the mocked `/quote` solver's solution
+    let mocked_quote_solution = mocked_solutsion(OrderUid([0u8; 56]));
+    mock_solver.configure_solution(Some(mocked_quote_solution));
+
+    let quote_request = OrderQuoteRequest {
+        from: bob.address(),
+        sell_token: dai.address(),
+        buy_token: onchain.contracts().weth.address(),
+        side: OrderQuoteSide::Sell {
+            sell_amount: SellAmount::AfterFee {
+                value: NonZeroU256::try_from(executed_amount).unwrap(),
+            },
+        },
+        ..Default::default()
+    };
+
+    let quote_response = services.submit_quote(&quote_request).await.unwrap();
```

Comment:
This is supposed to be the hand crafted quote, right?
I'm not sure how these asserts verify that this is the case. I think encoding an obviously custom value in the quote response and asserting that the value shows up in the quote would help.
Overall a comment explaining how the correctness of the key property of the test is asserted would be nice.

Reply:
> I think encoding an obviously custom value in the quote response and asserting that the value shows up in the quote would help.

To encode this somewhere inside mocked solutions? The returned quote contains limited data. Not sure how this can be achieved.

Reply:
The thing with the obviously custom value was just an idea. How do you currently know that the quote returned from the request actually came from the mock solution using the JIT order?
The main purpose of the e2e should be to verify that the new API can indeed be used to produce verified quotes based on JIT orders and so far it's not clear to me how to "prove" that the quote actually came from the mock solution.

Reply:
Ahh, now I get it. Completely forgot that we have separate CLI arguments for regular quoting and native price quoting. So since the `mock` solver is used for all the regular price quoting we know for sure that everything works as expected.
Played around with the test locally and indeed chanding tiny details about the cow amm order or quoted prices results in an error.
So I think this indeed indicates that everything is in order. 👍